### PR TITLE
ContributionConverter: Handle rn and version bump

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: https://github.com/demisto/etc/issues/29944

## Description
Added the update_type, release_notes, and detected_content_items to the ContributionConverter class.
Added support for update rn and version bump

## An example for RN file that contains modified files and new content items in the contrib:
https://github.com/demisto/content/pull/13682/files#diff-393157bde758d48d7fb5c030527abb92727a1425ec2f26a9a72df13b563bd685

## Must have
- [X] Tests

### CODE REVIEW

File: demisto_sdk/commands/update_release_notes/update_rn_manager.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def __init__(self, user_input: Optional[str] = None, update_type: Optional[str]
         # When a user choose a specific pack to update rn, the -g flag should not be passed
         if self.given_pack and self.is_all:
             raise ValueError('Please remove the -g flag when specifying only one pack.')
+        self.rn_path = None
```
[the remark this answers is not in the document]
Comment (Bargenish) [reply demisto_sdk/commands/update_release_notes/update_rn_manager.py:40]: Didnt take into consideration this scenario. thanks.

File: demisto_sdk/commands/init/contribution_converter.py [hunk 2]
```diff
@@ -479,3 +490,66 @@ def create_pack_metadata(data: Dict = {}) -> Dict:
...
+        entity_identifier = '##### '
+        template_text = '%%UPDATE_RN%%'
+
+        rn_per_content_item = self.format_user_input()
+
+        with open(rn_path, 'r+') as rn_file:
+            lines = rn_file.readlines()
+            for index in range(len(lines)):
+                if template_text in lines[index]:
+                    template_entity = lines[index - 1].lstrip(entity_identifier).rstrip('\n')
```
[the remark this answers is not in the document]
Comment (Bargenish) [reply demisto_sdk/commands/init/contribution_converter.py:548]: Not according to the template generation code of the *build_rn_desc* function.

File: demisto_sdk/commands/init/contribution_converter.py [hunk 3]
```diff
@@ -479,3 +490,66 @@ def create_pack_metadata(data: Dict = {}) -> Dict:
...
+                entity_name = UpdateRN.get_display_name(entity_path[0]) if entity_path else entity_name
+            elif not line.startswith(content_item_type_identifier):
+                rn_per_content_item[entity_name] = rn_per_content_item[entity_name] + line + '\n'
+        return rn_per_content_item
+
+    def replace_RN_template_with_value(self, rn_path: str):
+        """
+        Replace the release notes template for a given release-note file path
+        with the contributor's text.
+        Modify the template not overwrite it.
```
[the remark this answers is not in the document]
Comment (Bargenish) [reply demisto_sdk/commands/init/contribution_converter.py]: I updated the description.

File: demisto_sdk/commands/init/contribution_converter.py [hunk 4]
```diff
@@ -514,15 +514,17 @@ def format_user_input(self) -> Dict[str, str]:
...
         content_item_type_identifier = '#### '
-        rn = list(filter(None, self.release_notes.splitlines()))
         rn_per_content_item = defaultdict(str)
         entity_name = 'NonEntityRelated'
-        for line in rn:
+
+        items_path = {content_item.get('source_id'): content_item.get('source_file_name')
+                      for content_item in self.detected_content_items}
+
+        for line in list(filter(None, self.release_notes.splitlines())):
```
Comment (yaakovi) [inline demisto_sdk/commands/init/contribution_converter.py:523]:
```suggestion
        for line in filter(None, self.release_notes.splitlines()):
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**